source/MaterialXRuntime/RtStage.cpp

```diff
@@ -171,10 +173,16 @@ RtPrim RtStage::createNodeDef(RtNodeGraph& nodeGraph,
         nodedef.setNodeGroup(nodeGroup);
     }
 
-    // Add an output per nodegraph input
+    // Add an input per nodegraph input
     for (auto input : nodeGraph.getInputs())
     {
-        RtAttribute attr = nodedef.createInput(input.getName(), input.getType());
+        RtInput attr = nodedef.createInput(input.getName(), input.getType());
```

Comment:
Copy the nodegraph input uifolders to the nodedef inputs if there are any. This is needed for publishing.

Reply:
Again why just ui folder?

Reply:
Yes, you can copy all metadata instead. Accessible from the private API:

```
const PvtObject* obj = PvtObject::hnd(attr)->asA<PvtObject>();
const vector<RtToken>& metadataNames = obj->getMetadataOrder();
for (auto metadataName : metadataNames)
{
    const RtTypedValue* metadataValue = obj->getMetadata(metadataName);
    ... 
}
```